fil_fungible_token/src/runtime/messaging.rs

```diff
@@ -0,0 +1,160 @@
+    #[error("address not initialised: `{0}`")]
+    AddressNotInitialised(Address),
+    #[error("ipld serialization error: `{0}`")]
+    Ipld(#[from] IpldError),
+}
+
+/// An abstraction used to send messages to other actors
+pub trait Messaging {
+    /// Call the receiver hook on a given actor, specifying the amount of tokens pending to be sent
+    /// and the sender and receiver
+    ///
+    /// Returns true if the receiver hook is called and exits without error, else returns false
+    fn call_receiver_hook(
+        &self,
+        from: ActorID,
+        to: ActorID,
+        token_value: &TokenAmount,
+        data: &[u8],
+    ) -> Result<Receipt>;
+
+    /// Resolves the given address to it's ID address form
+    fn resolve_id(&self, address: &Address) -> Result<ActorID>;
+
+    ///  Creates an account at a pubkey address and returns the ID address
+    fn initialise_account(&self, address: &Address) -> Result<ActorID>;
+}
+
+#[derive(Debug, Default, Clone, Copy)]
+pub struct FvmMessenger {}
+
+impl Messaging for FvmMessenger {
+    fn call_receiver_hook(
+        &self,
+        from: ActorID,
+        to: ActorID,
+        token_value: &TokenAmount,
+        data: &[u8],
+    ) -> Result<Receipt> {
+        // TODO: use fvm_dispatch here (when it supports compile time method resolution)
+        // TODO: ^^ necessitates determining conventional method names for receiver hooks
+
+        // currently, the method number comes from taking the name as "TokensReceived" and applying
+        // the transformation described in https://github.com/filecoin-project/FIPs/pull/399
+        const METHOD_NUM: u64 = 1361519036;
+        let to = Address::new_id(to);
+
+        let params = TokenReceivedParams {
+            sender: Address::new_id(from),
+            value: token_value.clone(),
+            data: RawBytes::from(data.to_vec()),
+        };
+        let params = RawBytes::new(fvm_ipld_encoding::to_vec(&params)?);
+
+        Ok(send::send(&to, METHOD_NUM, params, TokenAmount::zero())?)
+    }
+
+    fn resolve_id(&self, address: &Address) -> Result<ActorID> {
+        actor::resolve_address(address).ok_or(MessagingError::AddressNotInitialised(*address))
+    }
+
+    fn initialise_account(&self, address: &Address) -> Result<ActorID> {
+        if let Err(e) = send::send(address, METHOD_SEND, Default::default(), TokenAmount::zero()) {
+            return Err(e.into());
+        }
+
+        actor::resolve_address(address).ok_or(MessagingError::AddressNotInitialised(*address))
+    }
+}
+
+/// A fake method caller
+///
+/// # call_receiver_hook
+/// If call_receiver_hook is called with an empty data array, it will return true.
+/// If call_receiver_hook is called with a non-empty data array, it will return false.
+///
+/// # resolve_id
+/// If Address is id-address or secp-address, it resolves by returing FAKE_RESOLVED_ID
+/// If Address is actor-address or bls-address, it returns an error simulating an uninitialised address
```

Comment:
I feel like there will make test-writing a little fiddly, and cause wide churn if we ever want to change these behaviours. All pushing towards either a much more robust fake, that can be configured, or a mock (#33).